Topic algorithms- exposure(rescale_intensity)

rnvikram opened the pull request:
Implemented the rescale intensity algorithm from the transform module with TensorFlow. The driver file can be run to test the algorithm. rescale_intensity from rescale_intensity must be called to use the algorithm.

shakes76 commented on transform/rescale_intensity/README.md:
```diff
@@ -0,0 +1,72 @@
+# Rescale Intensity
```
Good, but not dependencies list

shakes76 commented on transform/rescale_intensity/driver.py:
```diff
@@ -0,0 +1,71 @@
+import os
```
Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver/module does.

shakes76 commented on transform/rescale_intensity/driver.py:
```diff
@@ -0,0 +1,71 @@
+        np_output = rescale_intensity_numpy(array, out_range=out_range)
+        result = np.array_equal(np_output, tf_output)
+
+    if in_range != None and out_range != None:
```
Good to add inline comments even if original code didn't have them

shakes76 commented on transform/rescale_intensity/rescale_intensity.py:
```diff
@@ -0,0 +1,123 @@
+  sess = tf.InteractiveSession()
+  tf.global_variables_initializer().run()
+  imin, imax = intensity_range(input_image,dtype, in_range)
+  omin, omax = intensity_range(input_image,dtype,out_range,clip_negative=(imin >= 0))
```
More inline comments please

shakes76 commented:
## TF Usage
OK
## Good Practice (Design/Commenting)
OK
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
## Commit Log
Meaningful commit messages 
Progressive commits OK, more details next time
## Documentation
Description and Comments  good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required, changes to hard coded directories**
Request Description minimal. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info. Docstrings are important too.

shakes76 commented on transform/rescale_intensity/rescale_intensity.py:
```diff
@@ -0,0 +1,123 @@
+        range_values = dtype
+    if str(range_values) == 'image':
+        i_min = tf.reduce_min(image).eval()
+        i_max = tf.reduce_max(image).eval()
```
Good to add inline comments even if original code didn't have them

rnvikram commented:
Hello Shakes,
Thanks for reviewing the request.

1. **Commit log**- I had initially run all commits in the master branch. You can access it from here. https://github.com/rnvikram/PatternFlow/tree/master/rescale%20tensor

2. > Remove references to hard coded directories. Why is Kirti's directory in the code?
I used collab to implement the module. Since my machine did not have TensorFlow I used Kirti's machine to test the driver script. I had to make some changes. Looks like missed out removing the directory when transferring the code to my machine

3. > Good, but not dependencies list
I'm not sure about the comment. Does the feedback mean that I would have to include the dependencies used in the readme file?

4. > Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver/module does.
I've added the header to the tester file. Not sure if it should be added to the implementation. 

5. > Good to add inline comments even if original code didn't have them
Added inline comments as requested

6. > Good to add inline comments even if original code didn't have them
Added inline comments as requested


Need help with:
> Good, but not dependencies list
> Remove references to hard coded directories

Thanks
